### PR TITLE
TeleCine Presence

### PR DESCRIPTION
**Pull request by anaooz**
Adds "www.telecine.com.br", a brazilian streaming service for movies to the PreMiD system

### CODE REVIEW

File: websites/T/TeleCine/presence.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  browsing: "presence.playback.browsing"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {largeImageKey: "logo"}
+  const video: HTMLVideoElement = document.querySelector('body > div > div > div > div > div > video')
+
+  const path = document.location.pathname
+  
+  var pegarNome = () => {
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:16]: Merge vars

Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:16]: Use const

File: websites/T/TeleCine/presence.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    if(path.startsWith('/filmes/') && !path.endsWith('documentarios')){
+      const genero = document.querySelector('body > div > div > div > section > div > section > h1').textContent
+      const generoRegEx = /\s[A-Za-záàâãéèêíïóôõöúçÁÀÂÃÉÈÍÏÓÔÕÖÚÇ-]{3,}/
+      var regEx = new RegExp(generoRegEx)
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:20]: Use const

File: websites/T/TeleCine/presence.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      const generoNome = genero.match(regEx).shift()
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:21]: Merge vars

File: websites/T/TeleCine/presence.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  pause: "presence.playback.paused",
+  browsing: "presence.playback.browsing"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {largeImageKey: "logo"}
+  const video: HTMLVideoElement = document.querySelector('body > div > div > div > div > div > video')
+
+  const path = document.location.pathname
+
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:15]:
```suggestion
```

File: websites/T/TeleCine/presence.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+strings = presence.getStrings({
+  play: "presence.playback.playing",
+  pause: "presence.playback.paused",
+  browsing: "presence.playback.browsing"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {largeImageKey: "logo"}
+  const video: HTMLVideoElement = document.querySelector('body > div > div > div > div > div > video')
+
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:13]:
```suggestion
```

File: websites/T/TeleCine/presence.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  const path = document.location.pathname
+  
+  var pegarNome = () => {
+    if(path.startsWith('/filmes/') && !path.endsWith('documentarios')){
+      const genero = document.querySelector('body > div > div > div > section > div > section > h1').textContent
+      const generoRegEx = /\s[A-Za-záàâãéèêíïóôõöúçÁÀÂÃÉÈÍÏÓÔÕÖÚÇ-]{3,}/
+      var regEx = new RegExp(generoRegEx)
+      const generoNome = genero.match(regEx).shift()
+      return generoNome
+    }else if(path.includes('/filmes')){
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:23]:
```suggestion
    } else if(path.includes("/filmes")){
```

File: websites/T/TeleCine/presence.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      return undefined
+    }
+  }
+
+  switch (true) {
+    case path.endsWith("/mylist"):                 //LISTA
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:29]: Don't add all the spaces between the code and the comment

File: websites/T/TeleCine/presence.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      presenceData.details = "Visualizando lista..."
+    break
+
+    case path.includes("/franquias"):             //FRANQUIAS
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:33]: Ditto

File: websites/T/TeleCine/presence.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      presenceData.details = "Visualizando franquias..."
+    break
+
+    case path.includes("/franquia/"):
+      const franquia = document.querySelector('body > div > div > div > section > div > h1').textContent
+      presenceData.details = franquia
+      presenceData.state = "Visualizando a franquia..."
+    break
+
+    case path.includes("/cinelists"):           //CINELISTS
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:43]: Ditto

File: websites/T/TeleCine/presence.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      presenceData.details = "Visualizando cinelists..."
+    break
+
+    case path.includes("/cinelist/") || path.includes("/dc"):
+      const cinelist = document.querySelector('body > div > div > div > section > div > h1').textContent
+      presenceData.details = cinelist
+      presenceData.state = "Visualizando cinelist..."
+    break
+
+    case path.endsWith("documentarios"):            //DOCUMENTÁRIOS
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:53]: Ditto

File: websites/T/TeleCine/presence.ts [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+      presenceData.details = "Vendo documentários..."
+    break
+
+    case path.startsWith('/filmes/') && !path.endsWith('documentarios'):         //FILMES
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:57]: Ditto

File: websites/T/TeleCine/presence.ts [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+    case path.includes('/filmes'):
+      presenceData.details = "Vendo filmes por gênero..."
+    break
+
+    case path.startsWith('/filme/'):
+      const movie = document.querySelector('body > div > div > div > div > section > div > div > h1').textContent
+      presenceData.details = movie
+      presenceData.state = "Visualizando filme..."
+      if(!!video){
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:69]:
```suggestion
      if(video){
```

File: websites/T/TeleCine/presence.ts [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+          const filme = document.querySelector('body > div > div > div > div > section > div > div > h1').textContent
+          let timestamps = presence.getTimestampsfromMedia(video);
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:71]: Move this into the following if

File: websites/T/TeleCine/presence.ts [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+          presenceData.details = filme
+          presenceData.state = "Assistindo filme..."
+          presenceData.smallImageKey = video.paused ? "pause" : "play"
+          presenceData.smallImageText = video.paused ? (await strings).pause : (await strings).play
+        if (!video.paused) {
+            presenceData.startTimestamp = timestamps[0],
+            presenceData.endTimestamp = timestamps[1]
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:78]:
```suggestion
            [presenceData.startTimestamp, presenceData.endTimestamp] = presence.getTimestampsfromMedia(video)
```

File: websites/T/TeleCine/presence.ts [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+    case path.includes("/franquias"):             //FRANQUIAS
+      presenceData.details = "Visualizando franquias..."
+    break
+
+    case path.includes("/franquia/"):
+      const franquia = document.querySelector('body > div > div > div > section > div > h1').textContent
+      presenceData.details = franquia
+      presenceData.state = "Visualizando a franquia..."
+    break
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:41]: Wrap in brackets
```suggestion
    case path.includes("/franquia/"): {
      const franquia = document.querySelector(
        "body > div > div > div > section > div > h1"
      ).textContent;
      presenceData.details = franquia;
      presenceData.state = "Visualizando a franquia...";
      break;
    }
```

File: websites/T/TeleCine/presence.ts [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+    case path.includes("/cinelists"):           //CINELISTS
+      presenceData.details = "Visualizando cinelists..."
+    break
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:45]: Don't wrap in brackets
```suggestion
    case path.includes("/cinelists"): //CINELISTS
      presenceData.details = "Visualizando cinelists...";
      break;
    case path.includes("/cinelist/") || path.includes("/dc"): {
      const cinelist = document.querySelector(
        "body > div > div > div > section > div > h1"
      ).textContent;
      presenceData.details = cinelist;
      presenceData.state = "Visualizando cinelist...";
      break;
    }
```

File: websites/T/TeleCine/presence.ts [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {largeImageKey: "logo"}
+  const video: HTMLVideoElement = document.querySelector('body > div > div > div > div > div > video')
+
+  const path = document.location.pathname
+  
+  var pegarNome = () => {
+    if(path.startsWith('/filmes/') && !path.endsWith('documentarios')){
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:17]: Use double quotes

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Run `yarn lint` in the presence's folder - you will need to install eslint and prettier globally for this to work `npm install eslint prettier -g`

Address my requested cahnges as there's quite a few, once these changes have been addressed, we can work from there

File: websites/T/TeleCine/presence.ts (hunk 13 above)
Comment (anaooz) [reply websites/T/TeleCine/presence.ts:71]: that "state" doesn't belongs to the video section , they literally use the same url to the movie page (synopsis, actors, director,etc.) and the movie itself

**Comment by EncryptedDev** [comment]
Please rebase your fork, that will resolve DeepScan mostly

File: websites/T/TeleCine/dist/metadata.json [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+  "version": "1.0.0",
+  "color": "#071aa6",
+  "tags": [
+    "video",
+    "stream",
+    "media"
+  ],
+  "url": [
+    "www.telecine.com.br",
+    "telecine.com.br"
```
Comment (EncryptedDev) [inline websites/T/TeleCine/dist/metadata.json:19]: Website seems like it redirects to `www.*`

File: websites/T/TeleCine/presence.ts [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+  browsing: "presence.playback.browsing"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = { largeImageKey: "logo" };
+  const video: HTMLVideoElement = document.querySelector(
+    "body > div > div > div > div > div > video"
+  );
+  const path = document.location.pathname;
+  const pegarNome = () => {
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:16]: Merge consts

File: websites/T/TeleCine/presence.ts [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+    if (path.startsWith("/filmes/") && !path.endsWith("documentarios")) {
+      const genero = document.querySelector(
+        "body > div > div > div > section > div > section > h1"
+      ).textContent;
+      const generoRegEx = /\s[A-Za-záàâãéèêíïóôõöúçÁÀÂÃÉÈÍÏÓÔÕÖÚÇ-]{3,}/;
+      const regEx = new RegExp(generoRegEx);
+      const generoNome = genero.match(regEx).shift();
+      return generoNome;
+    } else if (path.includes("/filmes")) {
+      return undefined;
```
Comment (EncryptedDev) [inline websites/T/TeleCine/presence.ts:26]: Use conditional chaining instead

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/T/TeleCine/dist/metadata.json (hunk 18 above)
Comment (Slowlife01) [reply websites/T/TeleCine/dist/metadata.json:18]: https://github.com/PreMiD/Presences/pull/4892/commits/192607a8f9a7d515395483e3f7ab89edce547f38 why remove

**Comment by Slowlife01** [comment]
Upload proof of it working

**Comment by anaooz** [comment]
presence logo only working on main page, really don't know the issue; despite that, everything else is working
principal
principal2
rodando
rodando2
filme
filme2

**Comment by justarya** [comment]
> presence logo only showing on main page, really don't know if there is an issue; despite that, everything else is working principal principal2 rodando rodando2 filme filme2

I was looking through your PR. I thought I was the only person who got this problem.

I found out that you should be adding another asset in the discord developer portal. And make sure your assets matched with `largeImageKey` / `smallImageKey` in `presence.setActivity`

Screen Shot 2021-11-15 at 21 15 31

**Comment by anaooz** [comment]
thank you so much @justarya, now the presence is working correctly
l
l2
filme
filme2
franq
franq2

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Add a schema to your metadata

**Review by EncryptedDev — APPROVED** [review]
LGTM

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**